### PR TITLE
Bluetooth: host: Add non-connectable directed advertising support

### PR DESCRIPTION
**Pull request by joerchan**
This patch introduces two major changes to the directed advertising
feature of the bluetooth host.

Deprecating the bt_conn_create_slave_le, and removing
bt_conn_le_create_slave which has never been released. This behaviour
has now been moved by to providing the peer direct address into the
advertising parameters.

Introducing directed advertising support for nonconnectable
directed extended advertising, both scannable and non-scannable.

A bug was also fixed in the the directed-adv command in the shell
when the argument "low" was given. The advertiseng parameter pointer
declared with BT_LE_ADV_CONN_DIR_LOW_DUTY was declared in a scope that
was no longer valid when it was used to start the advertiser.

Add shell options to configure an extended advertiser for directed
advertising.
Add shell options to provide arbitrary advertising data as well as
giving scan response data.

### CODE REVIEW

**Comment by carlescufi** [comment]
Release notes please

**Comment by joerchan** [comment]
> Release notes please

Thanks. Added, also found a mistake with one of the last entries.

File: include/bluetooth/bluetooth.h [hunk 1]
```diff
@@ -450,13 +451,24 @@ struct bt_le_adv_param {
...
+	/** Directed advertising to peer
+	 *
+	 *  The advertising type will either be high duty cycle, or low duty
+	 *  cycle if the BT_LE_ADV_OPT_DIR_MODE_LOW_DUTY option is enabled.
+	 *
+	 *  In case of connectable high duty cycle if the connection could not
+	 *  be established within the timeout the connected() callback will be
+	 *  called with the status set to @ref BT_HCI_ERR_ADV_TIMEOUT.
+	 */
+	const bt_addr_le_t *peer;
```
Comment (carlescufi) [inline include/bluetooth/bluetooth.h:471]: Maybe add a comment describing when the stack will make use of this variable.

File: include/bluetooth/bluetooth.h [hunk 2]
```diff
@@ -465,45 +477,53 @@ struct bt_le_adv_param {
...
 
 /** @brief Start advertising
  *
  *  Set advertisement data, scan response data, advertisement parameters
  *  and start advertising.
  *
+ *  @note When starting directed advertising the advertisement data and scan
+ *        response data parameters are ignored. If the mode is high duty cycle
+ *        the timeout will be @ref BT_GAP_ADV_HIGH_DUTY_CYCLE_MAX_TIMEOUT.
+ *
```
Comment (carlescufi) [inline include/bluetooth/bluetooth.h:526]: Maybe add a note about the necessity to set `*peer` in the params here.

**Review by carlescufi — APPROVED** [review]
LGTM, just a couple of API comments

File: subsys/bluetooth/host/conn.c [hunk 3]
```diff
@@ -1553,6 +1553,27 @@ void bt_conn_process_tx(struct bt_conn *conn)
 	}
 }
 
+bool bt_conn_exists_le(u8_t id, const bt_addr_le_t *peer)
+{
+	struct bt_conn *conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, peer);
```
Comment (jhedberg) [inline subsys/bluetooth/host/conn.c:1558]: This function takes `u8_t id`, yet you pass `BT_ID_DEFAULT` to `bt_conn_lookup_addr_le()`. Is that intentional?

**Review by jhedberg — CHANGES_REQUESTED** [review]
See the one probable bug I found

File: subsys/bluetooth/host/conn.c (hunk 3 above)
Comment (joerchan) [reply subsys/bluetooth/host/conn.c:1558]: Not intentional, will fix. Thank you.

**Review by jhedberg — APPROVED** [review]

**Pull request merged**